### PR TITLE
Migration to Mkdocs platform

### PR DESCRIPTION
**Pull request by sarab97**
This commit migrates existing data to new platform Mkdocs with nice material theme.
The automated workflow has been included in this repo. This will build and deploy to gh-pages branch. 
Any existing workflow should be removed. 
A demo can be viewed on my repo.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +0,0 @@
-_site
```
Comment (kunalgrover05) [inline .gitignore:1]: You shouldn't be removing this file, rather use this to remove folders which you see on your local env that you shouldn't be committing.

File: LICENSE [hunk 2]
```diff
@@ -1,674 +0,0 @@
-                    GNU GENERAL PUBLIC LICENSE
```
Comment (kunalgrover05) [inline LICENSE]: This is critical! We need to have GPL v3 license in our repo

**Comment by kunalgrover05** [comment]
We are using Mkdocs, why does your PR say "Migration to Hugo"?

Some things are broken in your repo: 
- Do you know if we can honour the case in the Nav bar? eg: https://sarab97.github.io/debugging/debugging-mqtt/ should say `Debugging MQTT` instead of `Debugging mqtt`
- Some pages say "Index" as the title: https://sarab97.github.io/homeassistant/remote-access/ and https://sarab97.github.io/nvr-and-surveillance/zoneminder/
I suspect because the title is written in H2 instead of H1. Do you mind fixing it?
- Do you plan on doing a follow up PR for cleaning up pages and removing the top metadata with `parent` and stuff?
- (Unrelated to your change) Let's delete this incomplete doc for now: https://sarab97.github.io/esphome/esphome-power-monitoring-pzem004t/ 

Rest everything looks pretty much good to go.

**Comment by sarab97** [comment]
> We are using Mkdocs, why does your PR say "Migration to Hugo"?

Oh dam. Looks like I was high when writing the PR.

> * Do you know if we can honour the case in the Nav bar? eg: https://sarab97.github.io/debugging/debugging-mqtt/ should say `Debugging MQTT` instead of `Debugging mqtt`
> 
> * Some pages say "Index" as the title: https://sarab97.github.io/homeassistant/remote-access/ and https://sarab97.github.io/nvr-and-surveillance/zoneminder/

I planned those to be fixed after merge. As each document needs improvements. 



> (Unrelated to your change) Let's delete this incomplete doc for now: https://sarab97.github.io/esphome/esphome-power-monitoring-pzem004t/

Will Delete that :D

File: .gitignore (hunk 1 above)
Comment (sarab97) [reply .gitignore:1]: What i did was deleted everything in new branch and then recreated from scratch by just copying.
There arent anyfiles that needs to be in .gitignore for this repo. So i dont think we need to add one.

File: LICENSE (hunk 2 above)
Comment (sarab97) [reply LICENSE]: Gonna add it now

**Review by kunalgrover05 — APPROVED** [review]

**Pull request merged**